icicle/include/icicle/hash/poseidon.h

```diff
@@ -16,8 +16,43 @@ namespace icicle {
   // Poseidon constants are being initialized.
   template <typename S>
   struct PoseidonConstantsInitOptions {
-    // TODO: Define the struct with fields such as arity, alpha, nof_rounds, mds_matrix, etc.
+    unsigned int  arity;                  ///< Arity of a hash (number of inputs of the single hash).
+    unsigned int  alpha;                  ///< Sbox power.
+    unsigned int  partial_rounds;         ///< Number of upper full rounds of a single hash.
+    unsigned int  full_rounds_half;       ///< Number of partial rounds of a single hash.
+    S*            round_constants;        ///< Round constants (both of the full and the partial rounds). The order of the constants in the memory is according to the rounds order.
```

Comment:
is the matrices dims known and fixed? if yes, can't we simply define them and then copy the entire struct to GPU as is without an additional indirection